gymnasium/dev_wrappers/to_torch.py

```diff
@@ -0,0 +1,141 @@
+
+
+Device = Union[str, torch.device]
+
+
+@functools.singledispatch
+def torch_to_jax(value: Any) -> Any:
+    """Converts a PyTorch Tensor into a Jax DeviceArray."""
+    raise Exception(
+        f"No conversion for PyTorch to Jax registered for type: {type(value)}"
+    )
+
+
+@torch_to_jax.register(torch.Tensor)
+def _torch_to_jax(value: torch.Tensor) -> DeviceArray:
+    """Converts a PyTorch Tensor into a Jax DeviceArray."""
+    tensor = torch_dlpack.to_dlpack(value)  # pyright: ignore
+    tensor = jax_dlpack.from_dlpack(tensor)
+    return tensor
+
+
+@torch_to_jax.register(abc.Mapping)
+def _torch_mapping_to_jax(
+    value: Mapping[str, Union[torch.Tensor, Any]]
+) -> Mapping[str, Union[DeviceArray, Any]]:
+    """Converts a mapping of PyTorch Tensors into a Dictionary of Jax DeviceArrays."""
+    return type(value)(**{k: torch_to_jax(v) for k, v in value.items()})
+
+
+@torch_to_jax.register(abc.Iterable)
+def _torch_iterable_to_jax(
+    value: Iterable[Union[torch.Tensor, Any]]
+) -> Iterable[Union[DeviceArray, Any]]:
+    """Converts an Iterable from PyTorch Tensors to an iterable of Jax DeviceArrays."""
+    return type(value)(torch_to_jax(v) for v in value)
+
+
+@functools.singledispatch
+def jax_to_torch(value: Any, device: Device = None) -> Any:
+    """Converts a Jax DeviceArray into a PyTorch Tensor."""
+    raise Exception(
+        f"No conversion for Jax to PyTorch registered for type: {type(value)}"
+    )
+
+
+@jax_to_torch.register(DeviceArray)
+def _jax_to_torch(value: DeviceArray, device: Device = None) -> torch.Tensor:
+    """Converts a Jax DeviceArray into a PyTorch Tensor."""
+    dlpack = jax_dlpack.to_dlpack(value.astype("float32"))
+    tensor = torch_dlpack.from_dlpack(dlpack)
+    if device:
+        return tensor.to(device=device)
+    else:
+        return tensor
+
+
+@jax_to_torch.register(abc.Mapping)
+def _jax_mapping_to_torch(
+    value: Mapping[str, Union[DeviceArray, Any]], device: Device = None
+) -> Mapping[str, Union[torch.Tensor, Any]]:
+    """Converts a mapping of Jax DeviceArrays into a Dictionary of PyTorch Tensors."""
+    return type(value)(**{k: jax_to_torch(v, device) for k, v in value.items()})
+
+
+@jax_to_torch.register(abc.Iterable)
+def _jax_iterable_to_torch(
+    value: Iterable[Union[torch.Tensor, Any]]
+) -> Iterable[Union[DeviceArray, Any]]:
+    """Converts an Iterable from Jax DeviceArrays to an iterable of PyTorch Tensors."""
+    return type(value)(jax_to_torch(v) for v in value)
+
+
+class JaxToTorchV0(Wrapper):
+    """Wraps an environment so that it can be interacted with through PyTorch Tensors.
+
+    Actions must be provided as PyTorch Tensors and observations will be returned as PyTorch Tensors.
+
+    Note:
+        Extensive testing has not been done for handling the device that the Tensor is stored on as
+        well as managing the tensor's data type.
+    """
+
+    def __init__(self, env: Env, device: Optional[torch.device] = None):
```

Comment:
Should the type not be `Optional[Device]` using the definition to provide at the beginning